lua/luasnip/init.lua

```diff
@@ -63,24 +63,66 @@ local function default_snip_info(snip)
 	}
 end
 
-local function available(snip_info)
```

Comment:
Okay afaict this can also be achieved with the old available (scope=buf) and get_snippets (everything else). Even though this design is nice, separate functions are backward compatible, so I think we should stick with that

Reply:
quick note: my addition should be backwards compatible.
I'll get back with another reply later.

Reply:
Allright, so I think I understand what you mean but could you tell me a bit more specifically what you'd like me to do. Is my modification worth keeping in a modified state or should we scrap it completely?

Reply:
True, of course it is backward compatible :facepalm:
What I don't like about this is that there's no good reason for `snip_info` to not be a part of `opts`, except staying backward compatible.

I think we'll have to scrap it :/
An alternative would be to expose it in addition to `available` (and implement `available` using it), but all of its functionality can be covered better by `get_snippets` (since get_snippets can access the implementation of `snippet_collection`)

Reply:
Allright yeah I agree. I'll fix this.

Reply:
> An alternative would be to expose it in addition to available (and implement available using it), but all of its functionality can be covered better by get_snippets (since get_snippets can access the implementation of snippet_collection)

Yeah this makes sensr. I'll fix this.